sql/sql_alter.cc

```diff
@@ -498,6 +488,19 @@ bool Sql_cmd_alter_table::execute(THD *thd)
       thd->variables.auto_increment_offset = 1;
       thd->variables.auto_increment_increment = 1;
     }
```

Comment:
above assignments overwrite THD's auto increment settings. Are these variables returned somewhere to original values?

Reply:
Yes as comment before says at THD::reset_for_next_command()